packages/cli-kit/src/public/node/is-global.ts

```diff
@@ -1,18 +1,38 @@
+import {isUnitTest} from './context/local.js'
 import {PackageManager} from './node-package-manager.js'
 import {outputInfo} from './output.js'
+import {cwd, sniffForPath} from './path.js'
 import {captureOutput, exec, terminalSupportsRawMode} from './system.js'
 import {renderSelectPrompt} from './ui.js'
+import {execaSync} from 'execa'
+
+let _isGlobal: boolean | undefined
 
 /**
  * Returns true if the current process is running in a global context.
  *
- * @param env - The environment to check. Defaults to `process.env`.
+ * @param argv - The arguments passed to the process.
  * @returns `true` if the current process is running in a global context.
  */
-export function currentProcessIsGlobal(env = process.env): boolean {
-  // npm, yarn, pnpm and bun define this if run locally.
-  // If undefined, we can assume it's global (But there is no foolproof way to know)
-  return env.npm_config_user_agent === undefined
+export function currentProcessIsGlobal(argv = process.argv): boolean {
+  // If we are running tests, we need to disable the cache
+  if (_isGlobal !== undefined && !isUnitTest()) return _isGlobal
+
+  // Path where the current project is (app/hydrogen)
+  const path = sniffForPath() ?? cwd()
+
+  // Closest parent directory to contain a package.json file or node_modules directory
+  // https://docs.npmjs.com/cli/v8/commands/npm-prefix#description
+  const npmPrefix = execaSync('npm', ['prefix'], {cwd: path}).stdout.trim()
```

Comment:
You mentioned in the team Slack that this is more robust than looking at the environment variable -- could you expand a bit on why? Running an executable, even if quick, seems like it'd be more expensive.

Also, only tangently related -- how often do we call this? is it cached (in-memory)?